include/openthread/border_routing.h

```diff
@@ -482,6 +482,16 @@ void otBorderRoutingDhcp6PdSetEnabled(otInstance *aInstance, bool aEnabled);
  */
 otBorderRoutingDhcp6PdState otBorderRoutingDhcp6PdGetState(otInstance *aInstance);
 
+/**
+ * Function pointer callback which is invoked when state of request/release a prefix via DHCPv6 Prefix Delegation (PD)
+ * for Thread interface changed.
+ *
+ * @param[in] aContext A pointer to arbitrary context information.
+ *
+ */
+typedef void (*otBorderRoutingRequestDhcp6PdCallback)(bool aRequest, void *aContext);
+
+void otBorderRoutingSetRequestDhcp6PdCallback(otInstance *aInstance, otBorderRoutingRequestDhcp6PdCallback aCallback);
```

Comment:
Add doc for the new API?

Also may help to mention that "Subsequent calls to this function will replace the previously set callback".